File: 1807-evaluate-the-bracket-pairs-of-a-string/1807-evaluate-the-bracket-pairs-of-a-string.py
class Solution:
    def evaluate(self, s: str, knowledge: List[List[str]]) -> str:
        i = 0 
        sol = ""
        maps = {j[0]:j[1] for j in knowledge}
        while i < len(s):
            if s[i] == "(":
                i += 1
                copy = i
                temp = ""
                while s[i] != ")":
                    temp += s[i]
                    i+=1
                # print(temp, knowledge)
                if temp in maps:
                    sol += maps[temp]
                else:
                    sol += "?"
            else:
                sol += s[i]
            i+=1
        return sol

File: 1807-evaluate-the-bracket-pairs-of-a-string/1807-evaluate-the-bracket-pairs-of-a-string.cpp
class Solution {
public:
    string evaluate(string s, vector<vector<string>>& knowledge) {
        unordered_map<string, string> maps;
        for(auto i : knowledge)
            maps[i[0]] = i[1];
        int i = 0;
        string sol = "";
        while (i < s.size()){
            if (s[i] == '('){
                string temp = "";
                i++;
                while (s[i] != ')'){
                    temp += s[i];
                    i++;
                }
                if (maps.find(temp) != maps.end()){
                    sol += maps[temp];
                } else {
                    sol += "?";
                }
            } else {
                sol += s[i];
            }
            
            i++;
            
        }
        return sol;
    }
};